HKMP/Game/Client/Entity/MantisLord.cs

```diff
@@ -0,0 +1,907 @@
+                    // We might have never received the defeated event, but the entity is dead already
+                    if (healthManager.GetIsDead()) {
+                        _throneFsm.SetState("Pause");
+                    }
+                    else {
+                        _throneFsm.SetState("Wake");
+                    }
+                    break;
+                case ThroneAnimation.Defeated:
+                    _throneFsm.SetState("Start Sub");
+                    break;
+            }
+        }
+
+        public override void UpdateAnimation(byte animationIndex, byte[] animationInfo) {
+            if (animationIndex <= (byte) Animation.Throw2 || animationIndex == 255) {
+                base.UpdateAnimation(animationIndex, animationInfo);
+                return;
+            }
+
+            var animation = (ThroneAnimation) animationIndex;
+
+            _lastThroneAnimation = animation;
+
+            Logger.Get().Info(this, $"Received animation: {animation}");
+
+            if (animation == ThroneAnimation.Music) {
+                // Send the challenge event to stop the Deepnest gate from triggering
+                _challengePromptFsm.ExecuteActions("Send Challenge", 0);
+                _throneFsm.ExecuteActions("Music", 2, 3);
+            }
+
+            if (animation == ThroneAnimation.Stand) {
+                // Not sure if we can find some way to enable cam lock here.
+                // For now we don't as we don't want everyone in the scene to have cam lock...
+                _throneFsm.ExecuteActions("Stand", 1, 4);
+            }
+
+            if (animation == ThroneAnimation.Cage) {
+                // Same here with the area title, although this is not so annoying for the others
+                _throneFsm.ExecuteActions("Cage", 1, 2, 3, 4, 5);
+            }
+
+            if (animation == ThroneAnimation.Floors) {
+                _throneFsm.ExecuteActions("Floors", 1);
+            }
+
+            if (animation == ThroneAnimation.Leave) {
+                _throneFsm.ExecuteActions("Leave", 1, 2);
+            }
+
+            if (animation == ThroneAnimation.Wake) {
+                _throneFsm.ExecuteActions("Wake", 1, 2);
+            }
+
+            if (animation == ThroneAnimation.Defeated) {
+                _throneFsm.ExecuteActions("Defeated", 1, 2, 4);
+            }
+        }
+
+        private enum ThroneAnimation {
+            Music = Animation.Throw2 + 1,
+            Stand,
+            Cage,
+            Floors,
+            Leave,
+            Wake,
+            Defeated,
+        }
+
+        private enum ThroneState {
+            Idle = 0,
+            Stand,
+            Cage,
+            Floors,
+            Empty,
+            Defeated,
+        }
+    }
+
+    public abstract class MantisLordBase : HealthManagedEntity {
+        private readonly PlayMakerFSM _mantisFsm;
+        private Animation _lastAnimation;
+
+        public MantisLordBase(
```

Comment:
This constructor can also be `protected` as it shouldn't be instantiated from anything that doesn't inherit it.
```suggestion
        protected MantisLordBase(
```